imgui-binding/src/main/java/imgui/extension/imguizmo/ImGuizmo.java

```diff
@@ -0,0 +1,240 @@
+    private static final boolean[] buffer = new boolean[1];
+    private static final float[] matrixBuffer = new float[16];
+    private static final float[] positionBuffer = new float[3];
+    private static final float[] rotationBuffer = new float[3];
+    private static final float[] scaleBuffer = new float[3];
+
+    /*JNI
+       #include <imgui.h>
+       #include <ImGuizmo.h>
+       #include "jni_common.h"
+    */
+    private native static void nEnabled(boolean[] buffer); /*
+        ImGuizmo::Enable(&buffer[0]);
+    */
+
+    /**
+     * This will set the orthographic view
+     */
+    public static void setEnabled(boolean isEnabled) {
+        buffer[0] = isEnabled;
+        nEnabled(buffer);
+    }
+
+    private native static void nIsUsing(boolean[] buffer); /*
+        buffer[0] = ImGuizmo::IsUsing();
+    */
+
+    /**
+     * Natively checks to see if we're using the guizmo
+     */
+    public static boolean isUsing() {
+        nIsUsing(buffer);
+        return buffer[0];
+    }
+
+    private native static void nIsOver(boolean[] buffer); /*
+        buffer[0] = ImGuizmo::IsOver();
+    */
+
+    /**
+     * Natively checks to see if we're over the guizmo
+     */
+    public static boolean isOver() {
+        nIsOver(buffer);
+        return buffer[0];
+    }
+
+    private native static void nSetDrawList(long pointer); /*
+        ImGuizmo::SetDrawlist((ImDrawList*)pointer);
+    */
+
+    /**
+     * This will set the draw list of the given guizmo
+     */
+    public static void setDrawList(ImDrawList drawList) {
+        nSetDrawList(drawList.ptr);
+    }
+
+
+    /**
+     * This will set the default window drawlist value
+     */
+    public static void setDrawList() {
+        setDrawList(ImGui.getWindowDrawList());
+    }
+
+    private native static void nBeginFrame(); /*
+        ImGuizmo::BeginFrame();
+    */
+
+    /**
+     * Starts the next frame for the guizmo
+     */
+    public static void beginFrame() {
+        nBeginFrame();
+    }
+
+    private native static void nDecomposeMatrixToComponents(float[] matrix, float[] translation, float[] rotation, float[] scale); /*
+        ImGuizmo::DecomposeMatrixToComponents(matrix, translation, rotation, scale);
+    */
+
+    /**
+     * This will allow for manual editing of the guizmos
+     */
+    public static void decomposeMatrixToComponents(float[] matrix, float[] translation, float[] rotation, float[] scale) {
+
+        nDecomposeMatrixToComponents(matrixBuffer, positionBuffer, rotationBuffer, scaleBuffer);
+
+
+    }
+
+    private native static void nRecomposeMatrixFromComponents(float[] translation, float[] rotation, float[] scale, float[] matrix); /*
+        ImGuizmo::RecomposeMatrixFromComponents(translation, rotation, scale, matrix);
+    */
+
+
+    /**
+     * This will allow for manual editing of the guizmos
+     */
+    public static void recomposeMatrixFromComponents(float[] matrix, float[] translation, float[] rotation, float[] scale) {
```

Comment:
Arguments order should be the same as for native method. Not something randomly generated. 😕